rs/nns/governance/src/proposals/update_canister_settings.rs

```diff
@@ -338,4 +398,76 @@ mod tests {
             assert_eq!(update_canister_settings.valid_topic(), Ok(expected_topic),);
         }
     }
+
+    #[test]
+    fn test_update_canister_settings_to_self_describing() {
+        let update_canister_settings = UpdateCanisterSettings {
+            canister_id: Some(SNS_WASM_CANISTER_ID.get()),
+            settings: Some(CanisterSettings {
+                controllers: Some(Controllers {
+                    controllers: vec![ROOT_CANISTER_ID.get(), LEDGER_CANISTER_ID.get()],
+                }),
+                memory_allocation: Some(1 << 32),
+                wasm_memory_limit: Some(1 << 31),
+                wasm_memory_threshold: Some(1 << 30),
+                compute_allocation: Some(10),
+                freezing_threshold: Some(100),
+                log_visibility: Some(LogVisibility::Public as i32),
+            }),
+        };
+
+        let action = update_canister_settings.to_self_describing_action();
+        let value = SelfDescribingValue::from(action.value.unwrap());
+
+        assert_eq!(
+            value,
+            SelfDescribingValue::Map(hashmap! {
+                "canister_id".to_string() => SelfDescribingValue::Text(SNS_WASM_CANISTER_ID.get().to_string()),
+                "settings".to_string() => SelfDescribingValue::Map(hashmap! {
+                    "controllers".to_string() => SelfDescribingValue::Array(vec![
+                        SelfDescribingValue::Array(vec![
+                            SelfDescribingValue::Text(ROOT_CANISTER_ID.get().to_string()),
+                            SelfDescribingValue::Text(LEDGER_CANISTER_ID.get().to_string()),
+                        ]),
+                    ]),
+                    "memory_allocation".to_string() => SelfDescribingValue::Array(vec![SelfDescribingValue::Nat(candid::Nat::from(1u64 << 32))]),
```

Comment:
I probably mentioned this in another PR, but I think you could fairly easily simplify some of this, like so:

```
SelfDescribingValue::from(vec![1_u64 << 32])
```